rust_src/remacs-sys/lib.rs

```diff
@@ -18,159 +18,8 @@ extern crate libc;
-    Lisp_Int0 = 2,
-    Lisp_Int1 = 3 + (USE_LSB_TAG as usize as u8) * 3, // 3 | 6
-
-    // String.  XSTRING (object) points to a struct Lisp_String.
-    // The length of the string, and its contents, are stored therein.
-    Lisp_String = 4,
-
-    // Vector of Lisp objects, or something resembling it.
-    // XVECTOR (object) points to a struct Lisp_Vector, which contains
-    // the size and contents.  The size field also contains the type
-    // information, if it's not a real vector object.
-    Lisp_Vectorlike = 5,
-
-    // Cons.  XCONS (object) points to a struct Lisp_Cons.
-    Lisp_Cons = 6 - (USE_LSB_TAG as usize as u8) * 3, // 6 | 3
-
-    Lisp_Float = 7,
-}
-
-#[repr(isize)]
-#[derive(PartialEq, Eq, Copy, Clone, Debug)]
-pub enum PseudovecType {
-    PVEC_NORMAL_VECTOR = 0,
-    PVEC_FREE,
-    PVEC_PROCESS,
-    PVEC_FRAME,
-    PVEC_WINDOW,
-    PVEC_BOOL_VECTOR,
-    PVEC_BUFFER,
-    PVEC_HASH_TABLE,
-    PVEC_TERMINAL,
-    PVEC_WINDOW_CONFIGURATION,
-    PVEC_SUBR,
-    PVEC_OTHER,
-    PVEC_XWIDGET,
-    PVEC_XWIDGET_VIEW,
-    PVEC_THREAD,
-    PVEC_MUTEX,
-    PVEC_CONDVAR,
-    PVEC_MODULE_FUNCTION,
-
-    /* These should be last, check internal_equal to see why.  */
-    PVEC_COMPILED,
-    PVEC_CHAR_TABLE,
-    PVEC_SUB_CHAR_TABLE,
-    PVEC_RECORD,
-    PVEC_FONT, /* Should be last because it's used for range checking.  */
-}
-
-#[repr(C)]
-#[derive(PartialEq, Eq, Copy, Clone, Debug)]
-pub enum TextCursorKinds {
-    DEFAULT_CURSOR = -2,
-    NO_CURSOR = -1,
-    FILLED_BOX_CURSOR,
-    HOLLOW_BOX_CURSOR,
-    BAR_CURSOR,
-    HBAR_CURSOR,
-}
-
-// XXX: this can also be char on some archs
-pub type bits_word = size_t;
-
-/// Representation of an Emacs Lisp function symbol.
-#[repr(C)]
-pub struct Lisp_Subr {
-    pub header: Lisp_Vectorlike_Header,
-
-    /// This is the function pointer that will be called when the user invokes
-    /// the Emacs Lisp function. Also, this field is actually an union in C.
-    pub function: *const c_void,
-
-    /// The minimum number of arguments that can be passed to the Emacs Lisp
-    /// function.
-    pub min_args: c_short,
-
-    /// The maximum number of arguments that can be passed to te Emacs Lisp
-    /// function.
-    pub max_args: c_short,
-
-    /// The name of the function in Emacs Lisp.
-    pub symbol_name: *const c_char,
-
-    /// The interactive specification. This may be a normal prompt
-    /// string, such as `"bBuffer: "` or an elisp form as a string.
-    /// If the function is not interactive, this should be a null
-    /// pointer.
-    pub intspec: *const c_char,
-
-    // TODO: Change this to EMACS_INT
-    //
-    // If you wan't to give it a try and solve this you should see this commit:
-    // https://github.com/Wilfred/remacs/commit/c5461d03a411ff5c6f43885a0a9030e8a94bbc2e
-    /// The docstring of the Emacs Lisp function.
-    pub doc: *const c_char,
-}
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
```

Comment:
Yes, it has specific values for a few things that wouldn't otherwise be known. Could probably trim it slightly, we do end up with both EmacsInt and EMACS_INT which are the same thing.